fpdf/fpdf.py

```diff
@@ -4185,8 +4189,21 @@ def _raster_image(
         self.images_used_per_page_number[self.page].add(info["i"])
         return RasterImageInfo(**info, rendered_width=w, rendered_height=h)
 
+    def x_by_align(self, x, w, h, img_info, keep_aspect_ratio):
```

Comment:
Why did you move this out of `ImageInfo`?
(now that it actually uses another method from there...)

fpdf.py is already way too crowded. Specialized functionality should be delegated elsewhere whenever possible.

Reply:
> fpdf.py is already way too crowded. Specialized functionality should be delegated elsewhere whenever possible.

I agree.

However, the code of this method depends on one line on `img_info`, and instance of `ImageInfo`,
and on **three lines** on an instance of `FPDF`! (to access `.w`, `.r_margin`, `.l_margin`)

Given that this method is only invoked in two places, inside `FPDF._*_image()` methods,
it really did not make much sense to me to keep inside `ImageInfo`...

Reply:
I choosed to merge this PR as it is,
but I agree that we could try to move this method elswhere in another PR.

I'm not sure that `ImageInfo` is the best place though...

We could also have `FPDF` implement a `PageDimension` interface,
that provides `.w`, `.h`, `.epw`, `.eph`, `.t_margin` `.b_margin`, `.l_margin`, `.r_margin`...
This way `FPDF` instances could be provided to methods typed as `PageDimension`,
to reduce the coupling and make it explicit that only the page dimensions contained in `FPDF` instances is used.